core/lib/zksync_core/src/gas_tracker/mod.rs

```diff
@@ -103,9 +109,18 @@ pub(crate) fn commit_gas_count_for_l1_batch(
     let total_factory_deps_len: u32 = sorted_factory_deps
         .map(|factory_dep| factory_dep.len() as u32)
         .sum();
-    let additional_calldata_bytes = metadata.initial_writes_compressed.len() as u32
-        + metadata.repeated_writes_compressed.len() as u32
-        + metadata.l2_l1_messages_compressed.len() as u32
+
+    // Protocol Version 17 introduces boojum, and along with boojum an update to how storage writes
+    // are communicated/compressed.
+    let additional_calldata_bytes = match header.protocol_version.unwrap() {
+        ProtocolVersionId::Version17 => {
+            panic!("add in the size of the compressed state diffs from metadata")
```

Comment:
you panic here because we don't have `compressed_state_diffs` merged into main, right?

Reply:
anyway, the whole expression looks messy, please store the result of the match in a separate variable

Reply:
correct, i can update it. i tried out the match with addition and thought it was a clever way of chaining the two but dont feel strongly about keeping it that way